specification/purview/data-plane/Azure.Analytics.Purview.Scanning/stable/2023-09-01/scanningService.json

```diff
@@ -0,0 +1,9577 @@
+        }
+      }
+    },
+    "Scan": {
+      "description": "The scan.",
+      "required": [
+        "kind"
+      ],
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "kind": {
+          "enum": [
+            "AzureSubscriptionCredential",
+            "AzureSubscriptionMsi",
+            "AzureResourceGroupCredential",
+            "AzureResourceGroupMsi",
+            "AzureSynapseWorkspaceCredential",
+            "AzureSynapseWorkspaceMsi",
+            "AzureSynapseCredential",
+            "AzureSynapseMsi",
+            "AdlsGen1Credential",
+            "AdlsGen1Msi",
+            "AdlsGen2Credential",
+            "AdlsGen2Msi",
+            "AmazonAccountCredential",
+            "AmazonS3Credential",
+            "AmazonS3RoleARN",
+            "AmazonSqlCredential",
+            "AzureCosmosDbCredential",
+            "AzureDataExplorerCredential",
+            "AzureDataExplorerMsi",
+            "AzureFileServiceCredential",
+            "AzureSqlDatabaseCredential",
+            "AzureSqlDatabaseMsi",
+            "AzureSqlDatabaseNativeAuth",
+            "ArcEnabledSqlServerCredential",
+            "ArcEnabledSqlServerNativeAuth",
+            "AmazonPostgreSqlCredential",
+            "AzurePostgreSqlCredential",
+            "DatabricksCredential",
+            "SqlServerDatabaseCredential",
+            "AzureSqlDatabaseManagedInstanceCredential",
+            "AzureSqlDatabaseManagedInstanceMsi",
+            "AzureSqlDatabaseManagedInstanceNativeAuth",
+            "AzureSqlDataWarehouseCredential",
+            "AzureSqlDataWarehouseMsi",
+            "AzureMySqlCredential",
+            "HdfsCredential",
+            "AzureStorageCredential",
+            "AzureStorageMsi",
+            "TeradataTeradataCredential",
+            "TeradataTeradataUserPass",
+            "TeradataUserPass",
+            "OracleOracleCredential",
+            "OracleOracleUserPass",
+            "DatabricksHmsCredential",
+            "SapS4HanaSapS4HanaCredential",
+            "SapS4HanaSapS4HanaUserPass",
+            "SapEccSapEccCredential",
+            "SapEccSapEccUserPass",
+            "SnowflakeCredential",
+            "PowerBICredential",
+            "PowerBIDelegated",
+            "PowerBIFpa",
+            "PowerBIMsi",
+            "FabricCredential",
+            "FabricDelegated",
+            "FabricFpa",
+            "FabricMsi",
+            "TridentCredential",
+            "TridentDelegated",
+            "TridentMsi",
+            "DataverseCredential",
+            "DataverseMsi"
+          ],
+          "type": "string",
+          "description": "The properties of scan, vary by datasource type.",
+          "x-ms-enum": {
+            "name": "ScanAuthorizationType",
+            "modelAsString": true
+          }
+        },
+        "scanResults": {
```

Comment:
GitHub lost [my earlier comment](https://github.com/Azure/azure-rest-api-specs/pull/25983#discussion_r1380702007) so pasting an image here for context:

<img width="699" alt="image" src="https://github.com/Azure/azure-rest-api-specs/assets/85643503/82248fcf-3519-4b8b-96d1-b85feaa3de41">

Scan results are a separate "resource" from the REST API perspective, so I don't think they should be embedded within Scan resource like this.

A better design is to add a "top" parameter to the `ScanResult_ListScanHistory` so that the UI can request just the latest scan.  And that operation should return results "newest first" -- and document this in operation description.

Another option would be to just include the `id` of the most recent scan result within `Scan`, and then the UI could use this `id` in the `ScanResult_GetScanStatus` operation.

Reply:
@mikekistler Thanks for the detailed review on this. We do have a requirement that we want to return the scan with latest run status in a single call, the reason is that UX may need to show a list of scans and the latest run status in a one single page, if we separate to multiple calls, it will introduce some UX difference, say, we can't show the run status until all subsequent calls to get the run status return. Since we always return the single result, agree we can change it to a single result object instead of an array, thx.

Reply:
We changed the array into a single object: lastRunResult as discussed